File: Programing_Basics_with_JS/whileLoopMoreExercises/02.ReportSystem.js
function reportSystem(input) {
    let index = 0;
    let searchedMoney = Number(input[index]);
    index++;
    let command = input[index];
    index++;
    let payCount = 0;
    let productBuyWithCard = 0;
    let priceForProductBuyWithCard = 0;
    let productBuyWithCash = 0;
    let priceForProductBuyWithCash = 0;

    while(command !== 'End') {
        payCount++;
        let productPrice = Number(command);

        if(payCount % 2 === 0) {
            if(productPrice < 10) {
                console.log('Error in transaction!');
            } else {
                productBuyWithCard++;
                priceForProductBuyWithCard += productPrice;
                searchedMoney -= productPrice;
                console.log('Product sold!');
            }
        } else {
            if(productPrice > 100) {
                console.log('Error in transaction!');
            } else {
                productBuyWithCash++;
                priceForProductBuyWithCash += productPrice;
                searchedMoney -= productPrice;
                console.log('Product sold!');
            }
        }

        if(searchedMoney <= 0) {
            break;
        }

        command = input[index];
        index++;
    }

    let averagePaymentsWithCard = priceForProductBuyWithCard / productBuyWithCard;
    let averagePaymentsWithCash = priceForProductBuyWithCash / productBuyWithCash;

    if(searchedMoney <= 0) {
        console.log(`Average CS: ${averagePaymentsWithCash.toFixed(2)}`);
        console.log(`Average CC: ${averagePaymentsWithCard.toFixed(2)}`);
    } else {
        console.log('Failed to collect required money for charity.');
    }
}

reportSystem(['500', '120', '8', '63', '256', '78', '317'])
reportSystem(['600', '86', '150', '98', '227', 'End'])

File: Programming_Basics_with_Python/while_loop_more_exercises/02._report_system.py
needed_money = int(input())
command = input()

cash_payment = 0
cash_payment_count = 0
card_payment = 0
card_payment_count = 0
count = 0
enough_money = False

while not command == 'End':
    count += 1
    money = int(command)
    if count % 2 == 1:
        if money > 100:
            print('Error in transaction!')
        else:
            cash_payment += money
            cash_payment_count += 1
            needed_money -= money
            print('Product sold!')

    else:
        if money < 10:
            print('Error in transaction!')
        else:
            card_payment += money
            card_payment_count += 1
            needed_money -= money
            print('Product sold!')

    if needed_money <= 0:
        enough_money = True
        break

    command = input()
if enough_money:
    print(f'Average CS: {cash_payment / cash_payment_count:.2f}')
    print(f'Average CC: {card_payment / card_payment_count:.2f}')
else:
    print('Failed to collect required money for charity.')